File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution {
public:
    void Func(vector<int>& nums, vector<int> temp, vector<vector<int>>& res, int start)
    {
        res.push_back(temp);
        for(int i=start; i<nums.size(); i++)
        {
            if(i > start && nums[i] == nums[i - 1])
                continue;
            temp.push_back(nums[i]);
            Func(nums,temp,res,i+1);
            temp.pop_back();
        }
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<int>temp;
        vector<vector<int>> res;
        Func(nums, temp, res, 0);
        return res;
    }
};

File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    void Func(int[] nums, List<Integer> temp, List<List<Integer>> res, int start)
    {
        res.add(new ArrayList<>(temp));
        for(int i=start; i<nums.length; i++)
        {
            if(i > start && nums[i] == nums[i - 1])
                continue;
            temp.add(nums[i]);
            Func(nums, temp, res, i+1);
            temp.remove(temp.size() - 1);
        }
    }
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        Arrays.sort(nums);
        List<Integer> temp = new ArrayList<>();
        List<List<Integer>> res = new ArrayList<>();
        Func(nums, temp, res, 0);
        return res;
    }
}